x-pack/solutions/observability/plugins/apm/test/scout/ui/fixtures/page_objects/agent_keys.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaUrl, ScoutPage } from '@kbn/scout-oblt';
+
+export class AgentKeysPage {
+  constructor(private readonly page: ScoutPage, private readonly kbnUrl: KibanaUrl) {}
+
+  async goto() {
+    await this.page.goto(`${this.kbnUrl.app('apm')}/settings/agent-keys`);
+    await this.page.waitForLoadingIndicatorHidden();
+    await this.page
+      .getByRole('heading', { name: 'Settings', level: 1 })
+      .waitFor({ timeout: 10000 });
+
+    return this.page;
+  }
+
+  getCreateButtonLocator() {
+    return this.page.testSubj.locator('apmAgentKeysContentCreateApmAgentKeyButton');
+  }
+
+  async clickCreateButton() {
+    const button = this.getCreateButtonLocator();
+    await button.waitFor({ timeout: 5000 });
+    await button.click();
```

Comment:
Playwright should wait up to 10 sec by default, it makes sense if you want to cut it to 5 sec.